index.php

```diff
@@ -99,7 +114,7 @@ function do_blocks( $content ) {
 	global $registered_blocks;
 
 	// Extract the blocks from the post content
-	$open_matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)-->.*<!--\s*\/wp:\g1\s+-->/';
+	$open_matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)--><!--\s*\/wp:\g1\s+-->/';
```

Comment:
> Are there any cases we need to support that don't have tests?

We're covering everything for now I think. Oh maybe the register the same block twice.